MetaMask/metamask-mobile
# feat: upgrade assets controllers to v44

## **Description**

Upgrades the assets controllers to v44.  In this version, the token balances controller now stores erc20 balances across all chains and accounts, instead of just the current chain and account like before.  This allows polling erc20 balances across chains.

## **Related issues**

Depends on: https://github.com/MetaMask/metamask-mobile/pull/12340

## **Manual testing steps**

With PORTFOLIO_VIEW=false and PORTFOLIO_VIEW=true:

1. Onboard
2. Verify erc20 tokens have correct balances
3. Switch chains
4. Verify the erc20 tokens on the new chain have correct balances
5. Send a TX that changes an erc20 balance
6. Verify it gets updated after tx completes

## **Screenshots/Recordings**



### **Before**



### **After**



## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
